cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -0,0 +1,187 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/systemd"
+)
+
+var (
+	timeNow = time.Now
+
+	// default 1:30, as that is how long systemd will wait for services by
+	// default so seems a sensible default
+	defaultMountUnitWaitTimeout = time.Minute + 30*time.Second
+
+	unitFileDependOverride = `[Unit]
+Requires=%[1]s
+After=%[1]s
+`
+
+	doSystemdMount = doSystemdMountImpl
+)
+
+// SystemdMountOptions reflects the set of options for mounting something using
+// systemd-mount(1)
+type SystemdMountOptions struct {
```

Comment:
should this be exported? or just type aliased in export_test.go ?

Reply:
I suppose it could be type aliased in export_test.go